### PR TITLE
Add dnsmasq cache dump documentation

### PR DESCRIPTION
**Pull request by DL6ER**
Signed-off-by: DL6ER <dl6er@dl6er.de>

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [ ] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [ ] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [ ] I have tested my proposed changes, and have included unit tests where possible.
- [ ] I am willing to help maintain this change if there are issues with it later.
- [ ] I give this submission freely and claim no ownership.
- [ ] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [ ] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/contributing/).

---
**What does this PR aim to accomplish?:**
*A detailed description, screenshots (if necessary), as well as links to any relevant GitHub issues*

**How does this PR accomplish the above?:**
*A detailed description (such as a changelog) and screenshots (if necessary) of the implemented fix*

**What documentation changes (if any) are needed to support this PR?:**
*A detailed list of any necessary changes*

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Document the cache dump reports generated by `dnsmasq`. Despite https://www.mail-archive.com/dnsmasq-discuss@lists.thekelleys.org.uk/msg15294.html, no real documentation exists for this feature.

**How does this PR accomplish the above?:**

Add a docs page.

### CODE REVIEW

File: docs/extra.css [hunk 1]
```diff
@@ -14,7 +14,6 @@
 
 .md-typeset code,
 .md-typeset pre {
-    white-space: pre-wrap;
```
Comment (DL6ER) [inline docs/extra.css:17]: This is removed as it breaks the very wide output of the cache dump output into an unreadable format. This custom CSS was added some time ago [four years ago](https://github.com/pi-hole/docs/commit/959458a2a3f3a79da2c0758e6981f15f8d1e90e4) with no apparent reason (pinging @PromoFaux). I checked through the docs and did not find anything where this did something bad.

File: docs/ftldns/cache_dump.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+``` plain
+cache size 10000, 0/20984 cache insertions re-used unexpired cache entries.
+queries forwarded 10247, queries answered locally 14713
+queries for authoritative zones 0
+pool memory in use 22272, max 24048, allocated 480000
+server 127.0.0.1#5353: queries sent 10801, retried or failed 69
+server 192.168.2.1#53: queries sent 388, retried or failed 3
+
+Host                                     Address                        Flags      Expires
```
Comment (DL6ER) [inline docs/ftldns/cache_dump.md:19]: Another comment: The misplacement of the header to the lines is intended. I will fix it with a patch later when Simon is fully back (https://github.com/pi-hole/dnsmasq/pull/4).

File: docs/ftldns/cache_dump.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
+## Cache dump interpretation
+
+The `dnsmasq` core embedded into `pihole-FTL` prints a dump of the current cache content into the mail log file (default location `/var/log/pihole.log`) when receiving `SIGUSR1`, e.g. by
+
+``` bash
+sudo killall -USR1 pihole-FTL
+```
+
+Such a crash dump looks like
```
Comment (yubiuser) [inline docs/ftldns/cache_dump.md:9]:
```suggestion
Such a cache dump looks like
```

**Review by yubiuser — APPROVED** [review]

File: docs/extra.css (hunk 1 above)
Comment (PromoFaux) [reply docs/extra.css:17]: I cannot remember last week, let alone 4 years ago!

Comment (DL6ER) [reply docs/extra.css:17]: I assumed so. I figured it wouldn't hurt to ask, though.

**Pull request merged**